xaitk_saliency/impls/gen_descriptor_sim_sal/similarity_scoring.py

```diff
@@ -10,20 +10,14 @@ class SimilarityScoring (GenerateDescriptorSimilaritySaliency):
     """
     This saliency implementation transforms proximity in feature space into
     saliency heatmaps.
-    This should require a sequence of feature vectors of the query and reference
-    image, a number of feature vectors as predicted on perturbed versions of the
-    query image, as well as the masks of the query image perturbations (as would
-    be output from a `PerturbImage` implementation.
+    This should require feature vectors for the reference image, for each query
+    image, and for perturbed versions of the reference image, as well as the
+    masks of the query image perturbations (as would be output from a
```

Comment:
```suggestion
    image, perturbed versions of the reference image, the
    masks of the reference image perturbations (as would be output from a
```

Reply:
I think you need to add an `and` in the suggested sentence above.

Reply:
IMO, getting rid of the "and for" here makes it seem like the perturbed images themselves are required, not features vectors for those images. Similarly I think the "as well as" emphasizes that the required perturbation masks are not feature vectors.